tests/test_main.py

```diff
@@ -0,0 +1,43 @@
+import unittest
+from unittest.mock import patch, mock_open, MagicMock
+from repo_agent.chat_with_repo.chat_main import main
+
+class TestYourScript(unittest.TestCase):
+
+    def test_load_config(self):
+        mock_data = """
+        api_key: "12345"
+        api_base: "https://api.example.com"
+        db_path: "/path/to/db"
+        """
+        with patch("builtins.open", mock_open(read_data=mock_data)):
+            config = load_config("dummy_config.yml")
+            self.assertEqual(config['api_key'], "12345")
+            self.assertEqual(config['api_base'], "https://api.example.com")
+            self.assertEqual(config['db_path'], "/path/to/db")
+
+    @patch('your_script.RepoAssistant')
+    @patch('your_script.GradioInterface')
+    @patch('your_script.load_config')
```

Comment:
What does `your_script` means?

Reply:
it's the python file which contains the class, for RepoAssistant is rag.py etc

Reply:
Can you share test result?